datamodel/initdb.d/03_dcsa_im_v3_0.sql

```diff
@@ -515,35 +515,117 @@ CREATE TABLE dcsa_im_v3_0.shipment_transport (
 CREATE TABLE dcsa_im_v3_0.equipment_event (
+    equipment_event_type_code varchar(4) NOT NULL,
     equipment_reference varchar(15),
     empty_indicator_code text NOT NULL,
     transport_call_id uuid NOT NULL
 ) INHERITS (dcsa_im_v3_0.event);
 
 DROP TABLE IF EXISTS dcsa_im_v3_0.shipment_event CASCADE;
 CREATE TABLE dcsa_im_v3_0.shipment_event (
-    shipment_id uuid NOT NULL,
-    shipment_information_type_code varchar(3) NOT NULL
+    shipment_event_type_code varchar(4) NOT NULL,
+    shipment_information_type_code varchar(3) NOT NULL,
+    document_id varchar(50) NOT NULL,
+    reason varchar(100)
 ) INHERITS (dcsa_im_v3_0.event);
 
 DROP TABLE IF EXISTS dcsa_im_v3_0.transport_event CASCADE;
 CREATE TABLE dcsa_im_v3_0.transport_event (
+    transport_event_type_code varchar(4) NOT NULL,
     delay_reason_code varchar(3),
     vessel_schedule_change_remark varchar(250),
+    location_id uuid,
+    facility_type varchar(4),
     transport_call_id uuid NOT NULL
 ) INHERITS (dcsa_im_v3_0.event);
 
+DROP TABLE IF EXISTS dcsa_im_v3_0.operations_event CASCADE;
+CREATE TABLE dcsa_im_v3_0.operations_event (
+    operations_event_type_code varchar(4) NOT NULL, -- The code to identify the type of event that is related to the operation.
+    event_location uuid NOT NULL, -- The location where the event takes place.
+    port_call_service_type_code varchar(4) NOT NULL, -- The type of the service provided in the port call.
+    facility_type_code varchar(4) NULL -- Four character code to identify the specific type of facility.
+) INHERITS (dcsa_im_v3_0.event);
+
+DROP TABLE IF EXISTS dcsa_im_v3_0.port_call_service_type CASCADE;
+CREATE TABLE dcsa_im_v3_0.port_call_service_type (
+    port_call_service_type_code varchar(4) PRIMARY KEY, -- The 4-letter code indicating the type of the port call service.
+    port_call_service_type_name varchar(50) NOT NULL, -- The name of the of the port call service type.
+    port_call_service_type_description varchar(300) NOT NULL -- The description of the of the port call service type.
+);
+
+DROP TABLE IF EXISTS dcsa_im_v3_0.equipment_event_type CASCADE;
+CREATE TABLE dcsa_im_v3_0.equipment_event_type (
+    equipment_event_type_code varchar(4) PRIMARY KEY,
+    equipment_event_type_name varchar(30) NOT NULL
+);
+
+DROP TABLE IF EXISTS dcsa_im_v3_0.transport_event_type CASCADE;
+CREATE TABLE dcsa_im_v3_0.transport_event_type (
+    transport_event_type_code varchar(4) PRIMARY KEY,
+    transport_event_type_name varchar(30) NOT NULL
+);
+
+DROP TABLE IF EXISTS dcsa_im_v3_0.operations_event_type CASCADE;
+CREATE TABLE dcsa_im_v3_0.operations_event_type (
+    operations_event_type_code varchar(4) PRIMARY KEY,
+    operations_event_type_name varchar(30) NOT NULL,
+    operations_event_type_description varchar(200) NOT NULL
+);
+
+ALTER TABLE dcsa_im_v3_0.operations_event
+ADD FOREIGN KEY (event_classifier_code)
+REFERENCES dcsa_im_v3_0.event_classifier (event_classifier_code);
```

Comment:
Is this a good way to do it? My impression was that ShipmentEvents and TransportEvents can have different eventClassifierCodes.
Wouldn't a better way to add a CONSTRAINT in each of the event classes to only allow specific values?